### ISSUE TITLE
unable to drop out and back in  async_read_loop()

### ISSUE THREAD

**Issue by miccoli** [gvalkov/python-evdev#103]
I'm able to read events with the [async_read_loop](
https://python-evdev.readthedocs.io/en/latest/apidoc.html#evdev.eventio_async.EventIO.async_read_loop) API but I'm unable to stop and restart the `async for` loop. Here an example:

<details>
<summary>python example program</summary>

```python
import asyncio
import evdev

async def scan(devpath):
    dev = evdev.InputDevice(devpath)
    print("  * device opened")
    try:
        async for event in dev.async_read_loop():
            print("  event: {}".format(event))
    finally:
        dev.close()
        print("  * device closed")


async def amain():
    for i in range(3):
        print("Loop {}".format(i))
        task = asyncio.create_task(scan("/dev/input/event0"))
        await asyncio.sleep(10)
        task.cancel()
        while not task.done():
            await asyncio.sleep(0.1)

asyncio.run(amain())
```

</details>

This program creates a task that reads events from `/dev/input/event0`, waits 10 seconds, cancels the task and restarts it. The problem is that after the first iteration (Loop 0) the `dev.async_read_loop()` does not receive events any more.

<details>
<summary>Example output</summary>

```
Loop 0
  * device opened
  event: event at 1544047875.152415, code 04, type 04, val 786658
  event: event at 1544047875.152415, code 113, type 01, val 01
  event: event at 1544047875.152415, code 00, type 00, val 00
  event: event at 1544047875.174997, code 04, type 04, val 786658
  event: event at 1544047875.174997, code 113, type 01, val 00
  event: event at 1544047875.174997, code 00, type 00, val 00
  * device closed
Loop 1
  * device opened
  * device closed
Loop 2
  * device opened
  * device closed
```

</details>

Tested with evdev version 1.1.2, Python 3.7.1 on 'Linux 4.14.85-1-ARCH'

### edit ###
Same problem with all the [asyncio methods](
https://python-evdev.readthedocs.io/en/latest/apidoc.html#module-evdev.eventio_async)

**Comment by miccoli** [comment]
This bug is caused by the fact that the `loop.remove_reader` method is not called when the device is closed. 

Here a POC fix:
miccoli/python-evdev@1cd37402bd07b4125052b234699a0fa7be38ccf7

**Comment by gvalkov** [comment]
I pushed the proposed fix in c115d5b7745000d706a6d434888258a1a1ac20be. Sorry it took more than an year for me to come around ...
